IHP/ModelSupport.hs

```diff
@@ -578,14 +578,9 @@ logQuery query parameters time = do
 -- DELETE FROM projects WHERE id = '..'
 --
 -- Use 'deleteRecords' if you want to delete multiple records.
-deleteRecord :: forall model. (?modelContext :: ModelContext, Table model) => model -> IO ()
-deleteRecord model = do
-    let condition = primaryKeyCondition model
-    let whereConditions = condition |> map (\(field, _) -> field <> " = ?") |> intercalate " AND "
-    let theQuery = "DELETE FROM " <> tableName @model <> " WHERE " <> whereConditions
-    let theParameters = map snd condition
-    sqlExec (PG.Query . cs $! theQuery) theParameters
-    pure ()
+deleteRecord :: forall record id. (?modelContext :: ModelContext, Show id, Table record, HasField "id" record id, ToField id) => record -> IO ()
+deleteRecord record =
+    deleteRecordById (get #id record)
```

Comment:
Can you try passing the `record` type argument explicit here? This might help with the error

```
deleteRecordById @record (get #id record)
```